internal/provider/resource_gitlab_group_saml_link.go

```diff
@@ -0,0 +1,163 @@
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_group_saml_link", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_group_saml_link`" + ` resource allows to manage the lifecycle of an SAML integration with a group.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/groups.html#saml-group-links)`,
+
+		CreateContext: resourceGitlabGroupSamlLinkCreate,
+		ReadContext:   resourceGitlabGroupSamlLinkRead,
+		DeleteContext: resourceGitlabGroupSamlLinkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGitlabGroupSamlLinkImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Description: "The id of the GitLab group.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"access_level": {
+				Description:      fmt.Sprintf("Minimum access level for members of the SAML group. Valid values are: %s", renderValueListForDocs(validGroupSamlLinkAccessLevelNames)),
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validGroupSamlLinkAccessLevelNames, false)),
+				Required:         true,
+				ForceNew:         true,
+			},
+			"saml_group_name": {
+				Description: "The name of the SAML group.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"force": {
+				Description: "If true, then delete and replace an existing SAML link if one exists.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+			},
+		},
+	}
+})
+
+func resourceGitlabGroupSamlLinkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupId := d.Get("group_id").(string)
+	accessLevel := d.Get("access_level").(string)
+	samlGroupName := d.Get("saml_group_name").(string)
+	force := d.Get("force").(bool)
+
+	options := &gitlab.AddGroupSAMLLinkOptions{
+		AccessLevel:   &accessLevel,
+		SamlGroupName: &samlGroupName,
+	}
+
+	if force {
+		if err := resourceGitlabGroupSamlLinkDelete(ctx, d, meta); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] Create GitLab group SamlLink %s", d.Id())
+	SamlLink, _, err := client.Groups.AddGroupSAMLLink(groupId, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(buildTwoPartID(&groupId, &SamlLink.Name))
+
+	return resourceGitlabGroupSamlLinkRead(ctx, d, meta)
+}
+
+func resourceGitlabGroupSamlLinkRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	groupId := d.Get("group_id").(string)
+
+	// Try to fetch all group links from GitLab
+	log.Printf("[DEBUG] Read GitLab group SamlLinks %s", groupId)
+	samlLinks, _, err := client.Groups.ListGroupSAMLLinks(groupId, nil, gitlab.WithContext(ctx))
```

Comment:
The implementation here is missing pagination support. 

You should be able to support it using a code similar to this:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/375586b509db071c8aaf9f4f8b8f6d4ca6bd6359/internal/provider/resource_gitlab_deploy_token.go#L188

Reply:
Not paginated!